src/worker/services/workers.service.ts

```diff
@@ -16,10 +16,10 @@ export class WorkersService {
     if (!worker)
       throw new BadRequestException('Request body is required');
     try {
-      const newEmployee = new this.employeeModel(worker);
+      const newEmployee =  this.employeeModel.create(worker);
       const workerCode = this.generateUniqueNumber();
-      newEmployee.code = workerCode;
-      return await newEmployee.save();   
+      (await newEmployee).code = workerCode;
+      return  newEmployee;   
```

Comment:
תעבירי את ה await לשורה הראשונה
`      const newEmployee = await  this.employeeModel.create(worker);`